library/CMakeLists.txt

```diff
@@ -192,7 +192,7 @@ if(CMAKE_COMPILER_IS_GNUCC)
 endif(CMAKE_COMPILER_IS_GNUCC)
 
 if(CMAKE_COMPILER_IS_CLANG)
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wmissing-declarations -Wmissing-prototypes -Wdocumentation -Wno-documentation-deprecated-sync -Wunreachable-code")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wmissing-declarations -Wmissing-prototypes -Wno-documentation-deprecated-sync -Wunreachable-code")
```

Comment:
1. We mostly do want `-Wdocumentation`, it does point defects in our documentation. It's only problematic with Clang ≥15. So please add a version check, similar to what we're doing above with gcc options.
2. Please add a comment explaining why `-Wdocumentation` is problematic.